test_package/CMakeLists.txt

```diff
@@ -1,5 +1,6 @@
 project(test_package)
 cmake_minimum_required(VERSION 2.8.11)
+set(CMAKE_CXX_STANDARD 17)
```

Comment:
Is it strict necessary use C++17? Why not `set_property(TARGET ${PROJECT_NAME} PROPERTY CXX_STANDARD 11)`?

Reply:
Hi, I suppose it's not strictly necessary, since the lib will compile with C++11, however it does have some features that are enabled only for C++14 and C++17, so I thought it might be useful to exercise as much as possible in the test package.
Since I haven't set CXX_STANDARD_REQUIRED, the standard will "decay" to whatever is the latest supported by the toolchain in the Docker image.

Reply:
Understood.